src/Microsoft.ML.TimeSeries/SrCnnEntireTransformer.cs

```diff
@@ -0,0 +1,903 @@
+
+                var colType = inputSchema[_inputColumnIndex].Type;
+                if (!(colType is SrCnnTsPointDataViewType))
+                    throw Host.ExceptSchemaMismatch(nameof(inputSchema), "input", parent.InputColumnName, "SrCnnTsPoint", colType.ToString());
+
+                _parent = parent;
+                if (_parent.OutputLength == 3)
+                {
+                    _slotNames = new VBuffer<ReadOnlyMemory<char>>(_parent.OutputLength, new[] { "Is Anomaly".AsMemory(), "Raw Score".AsMemory(), "Mag".AsMemory() });
+                }
+                else if (_parent.OutputLength == 7)
+                {
+                    _slotNames = new VBuffer<ReadOnlyMemory<char>>(_parent.OutputLength, new[] { "Is Anomaly".AsMemory(), "Anomaly Score".AsMemory(), "Mag".AsMemory(),
+                        "Expected Value".AsMemory(), "Boundary Unit".AsMemory(), "Upper Boundary".AsMemory(), "Lower Boundary".AsMemory() });
+                }
+                else if (_parent.OutputLength == 4)
+                {
+                    _slotNames = new VBuffer<ReadOnlyMemory<char>>(_parent.OutputLength, new[] { "Is Anomaly".AsMemory(), "Raw Score".AsMemory(), "Mag".AsMemory(),
+                        "Expected Value".AsMemory() });
+                }
+            }
+
+            protected override DataViewSchema.DetachedColumn[] GetOutputColumnsCore()
+            {
+                var meta = new DataViewSchema.Annotations.Builder();
+                meta.AddSlotNames(_parent.OutputLength, GetSlotNames);
+                var info = new DataViewSchema.DetachedColumn[1];
+                info[0] = new DataViewSchema.DetachedColumn(_parent.OutputColumnName, new VectorDataViewType(NumberDataViewType.Double, _parent.OutputLength), meta.ToAnnotations());
+                return info;
+            }
+
+            public void GetSlotNames(ref VBuffer<ReadOnlyMemory<char>> dst) => _slotNames.CopyTo(ref dst, 0, _parent.OutputLength);
+
+            protected override Delegate MakeGetter(DataViewRow input, int iinfo, Func<int, bool> activeOutput, out Action disposer)
+            {
+                Contracts.AssertValue(input);
+                Contracts.Assert(0 <= iinfo && iinfo < _parent.ColumnPairs.Length);
+
+                var src = default(SrCnnTsPoint);
+                var getSrc = input.GetGetter<SrCnnTsPoint>(input.Schema[ColMapNewToOld[iinfo]]);
+
+                disposer = null;
+
+                ValueGetter<VBuffer<double>> del =
+                    (ref VBuffer<double> dst) =>
+                    {
+                        getSrc(ref src);
+                        if (src == null)
+                            return;
+                        var result = VBufferEditor.Create(ref dst, _parent.OutputLength);
+                        result.Values.Fill(Double.NaN);
+                        Double[] values = new Double[_parent.OutputLength];
+                        if (_parent.AnomalyModel.Inference(src, out values))
```

Comment:
>Inference [](start = 49, length = 9)

I am a bit confused by what this mapper does. The `Inference` method looks up an array of values in a dictionary where the keys are the timestamps seen during training. That means that for new data, it will return a vector of `NaN`s for any timestamp that was not seen during training (which is likely to be almost always). Is that the intention of your algorithm?

Reply:
> > Inference [](start = 49, length = 9)
> 
> I am a bit confused by what this mapper does. The `Inference` method looks up an array of values in a dictionary where the keys are the timestamps seen during training. That means that for new data, it will return a vector of `NaN`s for any timestamp that was not seen during training (which is likely to be almost always). Is that the intention of your algorithm?

Yes, your understanding is right.